lib/primitives/mtx.js

```diff
@@ -1122,12 +1136,31 @@ class MTX extends TX {
     assert(options, 'Options are required.');
     assert(options.changeAddress, 'Change address is required.');
 
+    // Ensure backward compatibility with the old API:
+    // Inputs to MTX were added with mtx.addOutpoint() and then later
+    // CoinSelector.fund() was expected to match those outpoints with coins
+    // from the wallet. The problem with this is that when CoinSelector
+    // initialized, it wiped out all the inputs, including some that may have
+    // already been signed outside the wallet for custom scripts or CoinJoins.
+    // With the new API, mtx.addCoin() can be used to either fund existing
+    // (pre-signed) inputs OR add new "preferred" inputs for covenants.
+    // The logic below was moved here from CoinSelector.fund() in case
+    // the old method was used and "preferred" coins are not yet in the view.
+    for (const input of this.inputs) {
```

Comment:
This will become totally unnecessary, if we do the thing mentioned above.

Also Complexity of this will be off charts. We already have issues with the `coins` being too big for the big wallets, and iterating it for every input will just explode )